src/System.Text.Json/src/System/Text/Json/Node/JsonObject.cs

```diff
@@ -0,0 +1,465 @@
+        /// </exception>
+        public bool ContainsProperty(string propertyName) => propertyName != null ? _dictionary.ContainsKey(propertyName) : throw new ArgumentNullException(nameof(propertyName));
+
+        /// <summary>
+        ///   Modifies the name of specified property.
+        /// </summary>
+        /// <param name="oldName">Old name of the property to change.</param>
+        /// <param name="newName">New name of the property to change.</param>
+        /// <exception cref="ArgumentException">
+        ///   Property name to set already exists if handling duplicates is set to <see cref="DuplicatePropertyNameHandling.Error"/>.
+        /// </exception>
+        /// <exception cref="KeyNotFoundException">
+        ///   Property with specified name is not found in JSON object.
+        /// </exception>
+        /// <remnark>
+        ///   Does not guarantee keeping the order the same.
+        /// </remnark>
+        public void ModifyPropertyName(string oldName, string newName)
+        {
+            if (!_dictionary.ContainsKey(oldName))
+                throw new KeyNotFoundException(string.Format(SR.PropertyNotFound, oldName));
+
+            JsonNode previousValue = _dictionary[oldName];
+            _dictionary.Remove(oldName);
+            _dictionary.Add(newName, previousValue);
+        }
+
+        /// <summary>
+        ///   Returns the value of a property with the specified name.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to return.</param>
+        /// <returns>Value of the property with the specified name.</returns>
+        /// <exception cref="KeyNotFoundException">
+        ///   Property with specified name is not found in JSON object.
+        /// </exception>
+        public JsonNode GetPropertyValue(string propertyName)
+        {
+            if (!TryGetPropertyValue(propertyName, out JsonNode jsonNode))
+            {
+                throw new KeyNotFoundException(string.Format(SR.PropertyNotFound, propertyName));
+            }
+
+            return jsonNode;
+        }
+
+        /// <summary>
+        ///   Returns the value of a property with the specified name.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to return.</param>
+        /// <param name="jsonNode">Value of the property with specified name.</param>
+        /// <returns>
+        ///  <see langword="true"/> if property with specified name was found;
+        ///  otherwise, <see langword="false"/>
+        /// </returns>
+        /// <remarks>
+        ///   When returns <see langword="false"/>, the value of <paramref name="jsonNode"/> is meaningless.
+        ///   Null <paramref name="jsonNode"/> doesn't mean the property value was "null" unless <see langword="true"/> is returned.
+        /// </remarks>
+        public bool TryGetPropertyValue(string propertyName, out JsonNode jsonNode)
+        {
+            if (!_dictionary.ContainsKey(propertyName))
```

Comment:
Use TryGetValue.

With the exception of the implementation of ContainsProperty method itself, you should never call ContainsKey.